pkg/rancher-desktop/main/extensions/extensions.ts

```diff
@@ -244,16 +244,7 @@ export class ExtensionImpl implements Extension {
         composeDir,
```

Comment:
Not part of this PR, but the locally-caught `throw` at lines 91 and 99 in this function mean that if there's no`result.icon` result, it looks like the generated error message will be:

`'Could not read extension metadata'`, (submessage: `'Invalid extension: missing icon'`) which I'm not sure you intended. Could be rewritten as:

```
    this._metadata ??= (async() => {
      const fallback = { vm: {} };

      try {
        const raw = await this.readFile('metadata.json');
        const result = _.merge({}, fallback, JSON.parse(raw));

        if (result.icon) {
          return result;
        }
      } catch (ex: any) {
        console.error(`Failed to read metadata for ${ this.id }: ${ ex }`);
        // Unset metadata so we can try again later
        this._metadata = undefined;
        throw new ExtensionErrorImpl(ExtensionErrorCode.INVALID_METADATA, 'Could not read extension metadata', ex);
      }
      throw new ExtensionErrorImpl(ExtensionErrorCode.INVALID_METADATA, 'Invalid extension: missing icon');
    })();
```